test/e2e/log_size_monitoring.go

```diff
@@ -0,0 +1,236 @@
+}
+
+func prepareData(masterAddress string, nodeAddresses []string) LogsSizeData {
+	data := make(map[string]map[string]*[]TimestampedSize)
+	ips := append(nodeAddresses, masterAddress)
+	for _, ip := range ips {
+		data[ip] = make(map[string]*[]TimestampedSize)
+		for _, path := range nodeLogsToCheck {
+			data[ip][path] = new([]TimestampedSize)
+		}
+		for _, path := range masterLogsToCheck {
+			data[ip][path] = new([]TimestampedSize)
+		}
+	}
+	return LogsSizeData{
+		data: data,
+		lock: sync.Mutex{},
+	}
+}
+
+func (d *LogsSizeData) AddNewData(ip, path string, timestamp time.Time, size int) {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	*d.data[ip][path] = append(
+		*d.data[ip][path],
+		TimestampedSize{
+			timestamp: timestamp,
+			size:      size,
+		},
+	)
+}
+
+func (d *LogsSizeData) PrintData() string {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	buf := &bytes.Buffer{}
+	w := tabwriter.NewWriter(buf, 1, 0, 1, ' ', 0)
+	fmt.Fprintf(w, "host\tlog_file\taverage_rate (B/s)\tnumber_of_probes\n")
+	for k, v := range d.data {
+		fmt.Fprintf(w, "%v\t\t\t\n", k)
+		for path, data := range v {
+			if len(*data) > 0 {
+				last := (*data)[len(*data)-1]
+				first := (*data)[0]
+				rate := (last.size - first.size) / int(last.timestamp.Sub(first.timestamp)/time.Second)
+				fmt.Fprintf(w, "\t%v\t%v\t%v\n", path, rate, len(*data))
+			}
+		}
+	}
+	w.Flush()
+	return buf.String()
+}
+
+// NewLogsVerifier creates a new LogsSizeVerifier which will stop when stopChannel is closed
+func NewLogsVerifier(c *client.Client, stopChannel chan bool) *LogsSizeVerifier {
+	nodeAddresses, err := NodeSSHHosts(c)
+	expectNoError(err)
+	masterAddress := getMasterHost() + ":22"
+
+	workChannel := make(chan WorkItem, len(nodeAddresses)+1)
+	workers := make([]*LogSizeGatherer, workersNo)
+
+	verifier := &LogsSizeVerifier{
+		client:        c,
+		stopChannel:   stopChannel,
+		data:          prepareData(masterAddress, nodeAddresses),
+		masterAddress: masterAddress,
+		nodeAddresses: nodeAddresses,
+		wg:            sync.WaitGroup{},
+		workChannel:   workChannel,
+		workers:       workers,
+	}
+	verifier.wg.Add(workersNo)
+	for i := 0; i < workersNo; i++ {
+		workers[i] = &LogSizeGatherer{
+			stopChannel: stopChannel,
+			data:        &verifier.data,
+			wg:          &verifier.wg,
+			workChannel: workChannel,
+		}
+	}
+	return verifier
+}
+
+// PrintData returns a string with formated results
+func (v *LogsSizeVerifier) PrintData() string {
+	return v.data.PrintData()
+}
+
+// Run starts log size gathering. It starts a gorouting for every worker and then blocks until stopChannel is closed
+func (v *LogsSizeVerifier) Run() {
+	v.workChannel <- WorkItem{
+		ip:    v.masterAddress,
+		paths: masterLogsToCheck,
+	}
+	for _, node := range v.nodeAddresses {
+		v.workChannel <- WorkItem{
+			ip:    node,
+			paths: nodeLogsToCheck,
+		}
+	}
+	for _, worker := range v.workers {
+		go worker.Run()
+	}
+	<-v.stopChannel
+	v.wg.Wait()
+
+	glog.Infof("\n%v", v.PrintData())
```

Comment:
At some point we decided to use Logf() in all our e2e tests to make it work better with ginkgo